server/game/gameSystems/SequentialSystem.ts

```diff
@@ -0,0 +1,73 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { GameEvent } from '../core/event/GameEvent';
+import { GameObject } from '../core/GameObject';
+import { GameSystem, IGameSystemProperties } from '../core/gameSystem/GameSystem';
+
+export interface ISequentialProperties extends IGameSystemProperties {
+    gameSystems: GameSystem[];
+}
+
+export class SequentialSystem extends GameSystem<ISequentialProperties> {
+    public constructor(gameSystems: GameSystem[]) {
+        super({ gameSystems: gameSystems });
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public override eventHandler() {}
+
+    public override queueGenerateEventGameSteps(events: GameEvent[], context: AbilityContext, additionalProperties = {}): void {
+        const properties = this.generatePropertiesFromContext(context, additionalProperties);
+        for (const gameSystem of properties.gameSystems) {
```

Comment:
Is the order of the gamesystems going to be consistent in all cases?

Reply:
Oh yeah I forgot to add a docstr on this system. The game systems will always execute sequentially. It's basically an implementation of the text "do [X], then do [Y], then do...".